strings/base_macros.h

```diff
@@ -45,6 +45,14 @@
 #undef _WINDOWS_NUMERICS_END_NAMESPACE_
 #endif
 
+#if defined(_MSC_VER)
+#define WINRT_IMPL_NOINLINE __declspec(noinline)
+#elif defined(__GNUC__)
+#define WINRT_IMPL_NOINLINE __attribute__((noinline))
+#else
+#define WINRT_IMPL_NOINLINE
+#endif
```

Comment:
How does this cover Clang? I believe clang already know how to interpret `__declspec(noinline)`. I wouldn't want to degrade Clang inadvertently.

Reply:
Clang defines `__GNUC__` and supports GNU-style `__attribute__((noinline))`. Clang has a (non-default) option to support MSVC-style `__declspec(noinline)` as well. This version removes the need to add the `-fdeclspec` flag when compiling with clang.

Reply:
The breakdown is

| Toolchain | Special macros | `__declspec(noinline)` | `__attribute__((noinline))`
|-|-|-|-|
| MSVC | `_MSC_VER` | ☑ | | |
| Clang (default) | `__GNUC__`<br>`__clang__` | | ☑ |
| Clang (`-fdeclspec`) | `__GNUC__`<br>`__clang__` | ☑ | ☑ |
| gcc | `__GNUC__` |  | ☑ |
| icc | `__GNUC__`<br>`__INTEL_COMPILER` |  | ☑ |
| icx | `__GNUC__`<br>`__INTEL_LLVM_COMPILER` | ☑ | ☑ |

djgpp, ellcc and zapcc are based on one of the above toolchains.

So sniffing for `_MSC_VER` and `__GNUC__` covers all the major toolchains.

Reply:
Great!